plutus-chain-index-core/src/Plutus/ChainIndex/Tx.hs

```diff
@@ -84,66 +86,67 @@ fromOnChainTx :: NetworkId -> OnChainTx -> ChainIndexTx
+                        { _citxTxId = txId tx
+                        , _citxInputs = map (fillTxInputWitnesses tx) txCollateral
+                        , _citxOutputs = InvalidTx
+                        , _citxValidRange = txValidRange
+                        , _citxData = txData
+                        , _citxRedeemers = calculateRedeemerPointers tx
+                        , _citxScripts = txScripts
+                        , _citxCardanoTx = Nothing
+                        }
+            )
             (fromOnChainCardanoTx False)
             ctx
 
+txRedeemersWithHash :: ChainIndexTx -> Map RedeemerHash Redeemer
+txRedeemersWithHash ChainIndexTx{_citxRedeemers} = Map.fromList
+    $ fmap (\r -> (redeemerHash r, r))
+    $ Map.elems _citxRedeemers
+
 -- Cardano api transactions store validity internally. Our emulated blockchain stores validity outside of the transactions,
 -- so we need to make sure these match up. Once we only have cardano api txs this can be removed.
 fromOnChainCardanoTx :: Bool -> SomeCardanoApiTx -> ChainIndexTx
 fromOnChainCardanoTx validity (SomeTx tx era) =
     either (error . ("Plutus.ChainIndex.Tx.fromOnChainCardanoTx: " ++) . show) id $ fromCardanoTx era $ setValidity validity tx
 
-mintingPolicies :: Map MintingPolicyHash MintingPolicy -> Map ScriptHash Script
-mintingPolicies = Map.fromList . fmap toScript . Map.toList
-  where
-    toScript (MintingPolicyHash mph, mp) = (ScriptHash mph, getMintingPolicy mp)
 
-validators :: [TxIn] -> (Map ScriptHash Script, Map DatumHash Datum, Redeemers)
-validators = foldMap (\(ix, txIn) -> maybe mempty (withHash ix) $ txInType txIn) . zip [0..] . sort
-  -- we sort the inputs to make sure that the indices match with redeemer pointers
-  where
-    -- TODO: the index of the txin is probably incorrect as we take it from the set.
-    -- To determine the proper index we have to convert the plutus's `TxIn` to cardano-api `TxIn` and
-    -- sort them by using the standard `Ord` instance.
-    withHash ix (ConsumeScriptAddress _lang val red dat) =
-      let (ValidatorHash vh) = validatorHash val
-       in ( Map.singleton (ScriptHash vh) (getValidator val)
-          , Map.singleton (datumHash dat) dat
-          , Map.singleton (RedeemerPtr Spend ix) red
-          )
-    withHash _ _ = mempty
+-- TODO: the index of the txin is probably incorrect as we take it from the set.
+-- To determine the proper index we have to convert the plutus's `TxIn` to cardano-api `TxIn` and
+-- sort them by using the standard `Ord` instance.
+calculateRedeemerPointers :: Tx -> Redeemers
+calculateRedeemerPointers tx = spends <> rewards <> mints <> certs
+    -- we sort the inputs to make sure that the indices match with redeemer pointers
 
-txRedeemersWithHash :: ChainIndexTx -> Map RedeemerHash Redeemer
-txRedeemersWithHash ChainIndexTx{_citxRedeemers} = Map.fromList
-    $ fmap (\r -> (redeemerHash r, r))
-    $ Map.elems _citxRedeemers
+    where
+        -- These sorts are ofc bad.
```

Comment:
okey, maybe not ofc. This applies https://github.com/input-output-hk/plutus-apps/pull/468#discussion_r945947243.

I am just unsure if in there we care if the `RedeemerPtr`'s match ones in ledger/node or only if theyre consistent across `plutus-apps`?

Reply:
I meant, what does the `ofc` acronym mean?

Reply:
sorry for confusion. ofc="of course" - as the ordering is different than it would be in cardano-ledger etc.

sorry for confusion, I only need to now if:
 1. assignment of RedeemerPtr's needs match the one from some formal spec
 2. assignment of RedeemerPtr's only needs to be consistent across plutus-apps code?
 
Also am I even right thinking the assignment of RedeemerPtr's for spending redeemers only satisfied 2.?

Reply:
@ak3n @sjoerdvisscher

Reply:
This is fine since this is only relevant for emulator Tx's.

Reply:
thats great